components/FilterForm/index.js

```diff
@@ -19,32 +19,45 @@ export default function FilterForm({ onFilterUpdate, plants }) {
   const filterPetFriendly = (plant, isPetFriendly) =>
     !isPetFriendly || plant.petFriendly === (isPetFriendly === "true");
 
-  const handleFilter = (event) => {
+  function handleSubmit(event) {
     event.preventDefault();
 
-    const newFilteredPlants = plants.filter(
-      (plant) =>
-        filterPlantSize(plant, plantSize) &&
-        filterSunlightRequirement(plant, sunlightRequirement) &&
-        filterWaterNeeds(plant, waterNeeds) &&
-        filterOptimalTemperature(plant, optimalTemperature) &&
-        filterPetFriendly(plant, petFriendly)
-    );
-
-    onFilterUpdate(newFilteredPlants);
-  };
-
-  function handleReset() {
-    onFilterUpdate(plants);
+    const newPreference = {
+      preferenceTitle: event.target.elements.title.value,
+      preferencePlants: plants.filter(function (plant) {
+        return (
+          filterPlantSize(plant, plantSize) &&
+          filterSunlightRequirement(plant, sunlightRequirement) &&
+          filterWaterNeeds(plant, waterNeeds) &&
+          filterOptimalTemperature(plant, optimalTemperature) &&
+          filterPetFriendly(plant, petFriendly)
+        );
+      }),
+    };
+
+    onAddPreference(newPreference);
```

Comment:
With this way you store the whole plant object as part of each preferences object. A better solution would be to add only the `id` of each plant to the `preferencePlants` array and not to copy the whole plant object into the preferences state. 
The reason is that this could lead to different sources of truth regarding your plant data. For example: What might happen if the user selects his preferred plants and the data is saved in the local storage for future references and afterwards you change your data and now you have different data. This can also lead to bugs we want to avoid. Therefore, you should solve it the same way as you did with your plant favorites. :) See more about this topic in our handout for [React State 3 ](https://github.com/neuefische/cgn-web-23-3/blob/main/sessions/react-state-3/react-state-3.md#choosing-the-state-structure).